test/pokemon/stab.test.ts

```diff
@@ -110,9 +113,80 @@ describe("STAB", () => {
     expect(enemyPokemon.calcStabMultiplierForTakingDamage).toHaveReturnedWith(1.5);
   });
 
-  it.todo("should have a 1.5 STAB on pledge moves");
+  it("combined Pledge moves should have a 1.5 STAB regardless of the user's type", async () => {
+    game.override.battleType("double");
 
-  it.todo("should have a 2.0 STAB on pledge moves if tera type DOES NOT match default type");
+    await game.classicMode.startBattle([SpeciesId.CHARMANDER, SpeciesId.SQUIRTLE]);
 
-  it.todo("should have a 2.25 STAB on pledge moves if tera type does match default type");
+    const enemyPokemon = game.scene.getEnemyField();
+    vi.spyOn(enemyPokemon[1], "calcStabMultiplierForTakingDamage");
+
+    game.setTurnOrder([BattlerIndex.PLAYER, BattlerIndex.PLAYER_2, BattlerIndex.ENEMY, BattlerIndex.ENEMY_2]);
+
+    // Charmander uses Water Pledge, then Squirtle uses Grass Pledge.
+    // The combined Pledge is Grass-type, but should still gain STAB.
+    game.move.use(MoveId.WATER_PLEDGE, 0, BattlerIndex.ENEMY);
+    game.move.use(MoveId.GRASS_PLEDGE, 1, BattlerIndex.ENEMY_2);
+
+    // Wait until both Pledges have been used
+    for (let i = 0; i < 2; i++) {
+      await game.phaseInterceptor.to("MoveEndPhase");
+    }
+
+    expect(enemyPokemon[1].calcStabMultiplierForTakingDamage).toHaveLastReturnedWith(1.5);
+  });
+
+  it("should have a 1.5 STAB on pledge moves if tera type DOES NOT match user's type", async () => {
+    game.override.battleType("double").startingHeldItems([{ name: "TERA_SHARD", type: ElementalType.WATER }]);
+
+    await game.classicMode.startBattle([SpeciesId.CHARMANDER, SpeciesId.SQUIRTLE]);
+
+    const playerPokemon = game.scene.getPlayerField();
```

Comment:
NIT: Use 
```suggestion
    const [playerPokemon] = game.scene.getPlayerField();
```
instead